lib/mongo/server/connection_pool.rb

```diff
@@ -668,17 +676,19 @@ def do_clear(options = nil)
             schedule_for_interruption(@checked_out_connections, service_id)
             schedule_for_interruption(@pending_connections, service_id)
           end
-        end
 
-        unless paused?
-          publish_cmap_event(
-            Monitoring::Event::Cmap::PoolCleared.new(
-              @server.address,
-              service_id: service_id,
-              interrupt_in_use_connections: options&.[](:interrupt_in_use_connections)
+          if @ready
+            publish_cmap_event(
+              Monitoring::Event::Cmap::PoolCleared.new(
+                @server.address,
+                service_id: service_id,
+                interrupt_in_use_connections: options&.[](:interrupt_in_use_connections)
+              )
             )
-          )
-          pause unless @server.load_balancer?
+            # Only pause the connection pool if the server was marked unknown,
+            # otherwise, allow the retry to be attempted with a ready pool.
+            do_pause if !@server.load_balancer? && @server.unknown?
```

Comment:
This is in response to the following test failure:
```
 Failure/Error: raise err
  
Mongo::Error::PoolClearedError:
   Connection to localhost:27018 interrupted due to server monitor timeout (for pool 0x856520) (modern retry, attempt 1, later retry failed: Mongo::Error::LintError: Server #<Server address=localhost:27018 PRIMARY replica_set=test-rs pool=#<ConnectionPool size=0 (0-3) used=0 avail=0 pending=0 paused>> is known but has non-ready pool)
```
We don't always mark the pool unknown for a SocketTimeoutError, so we should allow the pool to remain ready as long as the server remains known.